File: invision-backend/app/Enums/StoreCategory.php
<?php

namespace App\Enums;

enum StoreCategory: string
{
    case Supermarket = 'supermarket';
    case Hypermarket = 'hypermarket';
    case MiniMarket = 'mini_market';
    case Pharmacy = 'pharmacy';
    case ConvenienceStore = 'convenience_store';
    case WholesaleStore = 'wholesale_store';
    case Kiosk = 'kiosk';
    case Other = 'other';

    public function label(): string
    {
        return match ($this) {
            self::Supermarket => 'Supermarket',
            self::Hypermarket => 'Hypermarket',
            self::MiniMarket => 'Mini Market',
            self::Pharmacy => 'Pharmacy',
            self::ConvenienceStore => 'Convenience Store',
            self::WholesaleStore => 'Wholesale Store',
            self::Kiosk => 'Kiosk',
            self::Other => 'Other',
        };
    }
}

File: invision-mobile/lib/src/core/enums/store_category.dart
enum StoreCategory {
  supermarket,
  hypermarket,
  miniMarket,
  pharmacy,
  convenienceStore,
  wholesaleStore,
  kiosk,
  other;

  String get label => switch (this) {
        supermarket => 'Supermarket',
        hypermarket => 'Hypermarket',
        miniMarket => 'Mini Market',
        pharmacy => 'Pharmacy',
        convenienceStore => 'Convenience Store',
        wholesaleStore => 'Wholesale Store',
        kiosk => 'Kiosk',
        other => 'Other',
      };

  static StoreCategory fromString(String value) =>
      StoreCategory.values.firstWhere(
        (e) => e.name == value,
        orElse: () => StoreCategory.other,
      );
}
